File: snippets/react-native/list_payments.ts
import {
  listPayments,
  PaymentTypeFilter
} from '@breeztech/react-native-breez-sdk'

const exampleListPayments = async () => {
  // ANCHOR: list-payments
  try {
    const payments = await listPayments({})
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: list-payments
}

const exampleListPaymentsFiltered = async () => {
  // ANCHOR: list-payments-filtered
  try {
    const payments = await listPayments({
      filters: [PaymentTypeFilter.SENT],
      fromTimestamp: 1696880000,
      includeFailures: true
    })
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: list-payments-filtered
}

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ListPayments.kt
package com.example.kotlinmpplib

import breez_sdk.*
class ListPayments {
    fun listPayments(sdk: BlockingBreezServices) {
        // ANCHOR: list-payments
        try {
            val payments = sdk.listPayments(ListPaymentsRequest())
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: list-payments
    }

    fun listPaymentsFiltered(sdk: BlockingBreezServices) {
        // ANCHOR: list-payments-filtered
        try {
            val payments = sdk.listPayments(
                ListPaymentsRequest(
                    listOf(PaymentTypeFilter.SENT),
                    fromTimestamp = 1696880000,
                    includeFailures = true
                ))
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: list-payments-filtered
    }
}